libraries/load-themed-styles/src/index.ts

```diff
@@ -62,19 +93,41 @@ const _themeTokenRegex: RegExp = /[\'\"]\[theme:\s*(\w+)\s*(?:\,\s*default:\s*([
 /** Maximum style text length, for supporting IE style restrictions. */
 const MAX_STYLE_CONTENT_SIZE: number = 10000;
 
+const now: () => number =
+  () => (typeof performance !== 'undefined' && !!performance.now) ? performance.now() : Date.now();
+
+function measure(func: () => void): void {
+  const start: number = now();
+  func();
+  const end: number = now();
+  _themeState.perf.duration += end - start;
+}
+
 /**
  * Loads a set of style text. If it is registered too early, we will register it when the window.load
  * event is fired.
  * @param {string | ThemableArray} styles Themable style text to register.
  */
 export function loadStyles(styles: string | ThemableArray): void {
-  const styleParts: ThemableArray = Array.isArray(styles) ? styles : splitStyles(styles);
-
-  if (_injectStylesWithCssText === undefined) {
-    _injectStylesWithCssText = shouldUseCssText();
-  }
-
-  applyThemableStyles(styleParts);
+  measure(() => {
+    const styleParts: ThemableArray = Array.isArray(styles) ? styles : splitStyles(styles);
+    if (_injectStylesWithCssText === undefined) {
+      _injectStylesWithCssText = shouldUseCssText();
+    }
+    const {
+      mode,
```

Comment:
`const { mode = Mode.sync } = _themeState;` would avoid nullrefs and potential scenarios where old load-theme-styles loads first before a new version with this value.